Id Card signing on host (#32)
thorgate/django-esteid: esteid/idcard/README.md

# Signing with ID Card

## Process outline

1. Fetch the user certificate from the ID Card

   On the frontend, query the the `hwcrypto.js` API that interacts with the ID card,
   to obtain the certificate that would be required for the initial XAdEs structure.  

1. Initialization request to the backend 

   The backend creates a BDoc (Asic-E) container with files to be signed (or takes an existing container). 

>    Employing the user certificate received from the ID card on step 1, it creates a XAdEs structure 

Reviewer: ```suggestion
   Using the user certificate received from the ID card on step 1, it creates a XAdEs structure 
```

Author: I rephrased it a little to be even more clear :)

   containing the necessary data which constitute the value that is to be signed.
   This value is returned to the frontend.
   
   The container (if it is newly created), and the XAdEs structure, are stored in temporary files.
   
1. Signature generation (frontend) 

   The frontend passes the value for signing to the `hwcrypto.js` API, 
   and passes on the obtained signature to the backend. 
   
1. Finalization of the container (backend) 

   The backend picks up the XAdEs structure and container 
   from the temporary files created on step 2, 
   inserts the received signature into the XAdEs structure and packs it into the container.
  

>    The container should be saved to a desired location, and the temporary files cleaned up.

Reviewer: ```suggestion
   The container is saved to a desired location, and the temporary files are cleaned up.
```
